start.py

```diff
@@ -8,24 +11,71 @@
 
 from osfoffline.database import drop_db
 from osfoffline.gui.qt import OSFOfflineQT
-from osfoffline import settings
 from osfoffline.utils.log import start_logging
 from osfoffline.utils.singleton import SingleInstance
+from osfoffline import settings
+
+from updater4pyi import upd_source, upd_core
+from updater4pyi.upd_iface_pyqt4 import UpdatePyQt4Interface
+from updater4pyi.upd_defs import Updater4PyiError
 
+logger = logging.getLogger(__name__)
 
 if settings.DEBUG:
     signal.signal(signal.SIGINT, signal.SIG_DFL)
 
 
-def running_warning():
+def running_warning(message=None, critical=False):
+    if not message:
+        message = 'OSF-Offline is already running. Check out the system tray.'
     warn_app = QApplication(sys.argv)
-    QMessageBox.information(None, 'Systray', 'OSF-Offline is already running. Check out the system tray.')
+    if critical:
+        QMessageBox.critical(None, 'Systray', message)
+    else:
+        QMessageBox.information(None, 'Systray', message)
     warn_app.quit()
 
 
 def start():
     start_logging()
-    singleton = SingleInstance(callback=running_warning)  # will end application if an instance is already running
+
+    # will end application if an instance is already running
+    singleton = SingleInstance(callback=running_warning)
```

Comment:
It feels a bit clumsy to pass a 'callback' to this class's constructor. Really this is an error handler and not a callback.

It may be nicer to just try/except the instantiation of the class and to keep this kind of error handling explicitly in the scope of the start function.